src/components/Audioplayer/index.js

```diff
@@ -24,29 +26,37 @@ const styles = require('./style.scss');
 @connect(
   state => ({
     file: state.audioplayer.file,
+    surahs: state.surahs.entities,
     qari: state.audioplayer.qari,
     surah: state.audioplayer.surah,
     progress: state.audioplayer.progress,
     duration: state.audioplayer.duration,
     currentTime: state.audioplayer.currentTime,
     isPlaying: state.audioplayer.isPlaying,
     shouldRepeat: state.audioplayer.shouldRepeat,
+    shouldContinuous: state.audioplayer.shouldContinuous,
+    shouldRandom: state.audioplayer.shouldRandom,
   }),
-  { load, play, pause, playPause, update, repeat, next, previous }
+  { load, play, pause, playPause, update, repeat, next, previous, continuous, random }
 )
 export default class Audioplayer extends Component {
   static propTypes = {
     load: PropTypes.func.isRequired,
+    surahs: PropTypes.object.isRequired,
```

Comment:
Actually this is harder than I expected, surahs is an object of all 114 surahs. I added Shape on the surah which is more realistic.